hack/kubernetes_version/update_kubernetes_version.go

```diff
@@ -14,159 +14,442 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+/*
+The script expects the following env variables:
+ - UPDATE_TARGET=<string>: optional - if unset/absent, default option is "fs"; valid options are:
+   - "fs"  - update only local filesystem repo files [default]
+   - "gh"  - update only remote GitHub repo files and create PR (if one does not exist already)
+   - "all" - update local and remote repo files and create PR (if one does not exist already)
+ - GITHUB_TOKEN=<string>: The Github API access token. Injected by the Jenkins credential provider.
+   - note: GITHUB_TOKEN is needed only if UPDATE_TARGET is "gh" or "all"
+*/
+
 package main
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"flag"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
+	"text/template"
 	"time"
 
-	"github.com/golang/glog"
+	"golang.org/x/oauth2"
+
+	"k8s.io/klog/v2"
```

Comment:
I recommend keep using glog till we deprecate glog (that is a WIP PR it is better to let that PR handle migration from glog to klog)

Reply:
ok, thanks @medyagh for your review and comment - i've switched back from using klog to glog